src/flashbar/collapsible-flashbar.tsx

```diff
@@ -140,16 +147,28 @@ export default function CollapsibleFlashbar({ items, ...restProps }: FlashbarPro
 
     if (initialAnimationState) {
       updateBottomSpacing();
+
+      // When collapsing, scroll up if necessary to avoid losing track of the focused button
+      const shouldScrollUp =
+        notificationBarRef.current &&
+        !isFlashbarStackExpanded &&
+        isElementTopBeyondViewport(notificationBarRef.current);
+
       animate({
         elements: getElementsToAnimate(),
         oldState: initialAnimationState,
         newElementInitialState: ({ top }) => ({ scale: 0.9, y: -0.2 * top }),
         onTransitionsEnd: () => setTransitioning(false),
       });
+
+      if (notificationBarRef.current && shouldScrollUp) {
+        scrollElementIntoView(notificationBarRef.current);
+      }
+
       setTransitioning(true);
       setInitialAnimationState(null);
     }
-  }, [updateBottomSpacing, getElementsToAnimate, initialAnimationState, isFlashbarStackExpanded]);
+  }, [updateBottomSpacing, getElementsToAnimate, initialAnimationState, isFlashbarStackExpanded, isFocusVisible]);
```

Comment:
Dependency on `isFocusVisible` is no longer needed

Reply:
Right, removed